packages/event-status-service/src/controllers/automation/create.ts

```diff
@@ -0,0 +1,53 @@
+import { Response } from 'express'
+import uuidv4 from 'uuid/v4'
+import { logger } from '../../core/logger'
+import { AppRequest, stringifyError } from '@reapit/node-utils'
+import { db } from '../../core/db'
+import { Automation, generateAutomationItem } from '../../schemas/automation.schema'
+
+type Payload = {
+  clientCode: Automation['clientCode']
```

Comment:
Feels like we are losing some strong typing here... I am guessing TS doesn't like Automation.clientCode because it's not a class instance. Would it make sense to do something like;

```ts
interface Foo {
    // blah blah
}

class Automation implements Foo {
   // blah
}
```
Then Payload can be replaced by Foo and we are typesafe. Can explain if this doesn't make sense or if I am missing something :-)